plinth/modules/matrixsynapse/views.py

```diff
@@ -74,3 +79,24 @@ def get_context_data(self, *args, **kwargs):
         context = super().get_context_data(**kwargs)
         context['domain_name'] = matrixsynapse.get_configured_domain_name()
         return context
+
+    def get_initial(self):
+        """Return the values to fill in the form."""
+        initial = super().get_initial()
+        initial.update({
+            'enable_public_registration': get_public_registration_status()})
+        return initial
+
+    def form_valid(self, form):
+        """Handle valid form submission."""
+        enable_registration = form.cleaned_data['enable_public_registration']
+        if enable_registration:
```

Comment:
Checking here only for enable/disable is not enough, I think.
I think messages for enabled/disabled should be shown only if the form *changed*, like for the other forms. In case nothing changed, that should be made clear with an appropriate message as well (like in the other modules). You can have a look into the ejabberd form, the logic there is very similar.